src/main/java/edu/harvard/iq/dataverse/api/Users.java

```diff
@@ -98,4 +101,86 @@ public Response changeAuthenticatedUserIdentifier(@PathParam("identifier") Strin
         return ok("UserIdentifier changed from " + oldIdentifier + " to " + newIdentifier);
     }
     
+    @Path("token")
+    @DELETE
+    public Response deleteToken() {
+        User u;
+
+        try {
+            u = findUserOrDie();
+        } catch (WrappedResponse ex) {
+            return ex.getResponse();
+        }
+        AuthenticatedUser au;        
+       
+        try{
+             au = (AuthenticatedUser) u; 
+        } catch (ClassCastException e){ 
+            //if we have a non-authentivated user we stop here.
+            return notFound("Token for " + u.getIdentifier() + " not found.");
+        }       
+       
+        if (au == null) {
+            return notFound("Token for " + u.getIdentifier() + " not found.");
+        }
+        
+        authSvc.removeApiToken(au);
+        return ok("Token for " + au.getUserIdentifier() + " deleted.");
+        
+    }
+    
+    @Path("token")
+    @GET
+    public Response getTokenExpirationDate() {
+        User u;
+
+        try {
+            u = findUserOrDie();
+        } catch (WrappedResponse ex) {
+            return ex.getResponse();
+        }      
+        
+        ApiToken token = authSvc.findApiToken(getRequestApiKey());
+        
+        if (token == null) {
+            return notFound("Token " + getRequestApiKey() + " not found.");
+        }
+        
+        return ok("Token " + getRequestApiKey() + " expires on " + token.getExpireTime());
+        
+    }
+    
+    @Path("token/recreate")
+    @POST
+    public Response recreateToken() {
+        User u;
+
+        try {
+            u = findUserOrDie();
+        } catch (WrappedResponse ex) {
+            return ex.getResponse();
+        }
+        
+        AuthenticatedUser au;        
+        try{
+             au = (AuthenticatedUser) u; 
+        } catch (ClassCastException e){ 
+            //if we have a non-authentivated user we stop here.
+            return notFound("Token for " + u.getIdentifier() + " not found.");
```

Comment:
Same as above, this message could say that the Guest user can't have an API token. Or PrivateURL user, I guess. I'm just thinking that it might be nice to have a slightly different message than the one below.